services/repository/mongodb/service.go

```diff
@@ -0,0 +1,173 @@
+		err = client.Ping(context.TODO(), nil)
+		if err != nil {
+    		return nil, repository.NewUnknownErrorWithError("database is not accessible.", err)
+		}
+
+		//Todo : Do we need to have GetDbName in configuration service ? for now it is just hard coded
+		collection := client.Database("tenantdb").Collection("tenants")
+
+		filter := bson.D{{"TenantID", request.TenantID}}
+		var tenant models.Tenant
+
+		err = collection.FindOne(context.TODO(), filter).Decode(&tenant)
+		if err != nil {
+			return nil, repository.NewTenantNotFoundError(request.TenantID)
+		}
+
+		return &repository.ReadTenantResponse{Tenant: tenant}, nil
+	}
+
+
+// CreateTenant creates a new tenant.
+// context: Optional The reference to the context
+// request: Mandatory. The request to create a new tenant
+// Returns either the result of creating new tenant or error if something goes wrong.
+func (service *mongodbRepositoryService) CreateTenant(
+	ctx context.Context,
+	request *repository.CreateTenantRequest) (*repository.CreateTenantResponse, error) {
+
+	clientOptions := options.Client().ApplyURI(service.dbConnectionString)
+	client, err := mongo.Connect(context.TODO(), clientOptions)		
+	if err != nil {
+		return nil, repository.NewUnknownErrorWithError("Could not connect to mongodb database.", err)
+	}
+
+	//Todo : make sure if ping is necessary
+	err = client.Ping(context.TODO(), nil)
+	if err != nil {
+		return nil, repository.NewUnknownErrorWithError("database is not accessible.", err)
+	}
+
+ 	//Todo : Do we need to have GetDbName in configuration service ? for now it is just hard coded
+	collection := client.Database("tenantdb").Collection("tenants")
+	insertResult, err := collection.InsertOne(context.TODO(), request.Tenant)
+
+	if err != nil {
+    	return nil, repository.NewUnknownErrorWithError("Insert tenant failed.", err)
+	}
+
+	return &repository.CreateTenantResponse{
+		TenantID: insertResult.InsertedID.(string),
+	}, nil
+}
+
+
+// UpdateTenant update an existing tenant
+// context: Optional The reference to the context
+// request: Mandatory. The request to update an existing tenant
+// Returns either the result of updateing an exiting tenant or error if something goes wrong.
+func (service *mongodbRepositoryService) UpdateTenant(
+	ctx context.Context,
+	request *repository.UpdateTenantRequest) (*repository.UpdateTenantResponse, error) {
+
+	clientOptions := options.Client().ApplyURI(service.dbConnectionString)
+	client, err := mongo.Connect(context.TODO(), clientOptions)		
+	if err != nil {
+		return nil, repository.NewUnknownErrorWithError("Could not connect to mongodb database.", err)
+	}
+	
+	//Todo : make sure if ping is necessary
+	err = client.Ping(context.TODO(), nil)
+	if err != nil {
+		return nil, repository.NewUnknownErrorWithError("database is not accessible.", err)
+	}
+	
+	 //Todo : Do we need to have GetDbName in configuration service ? for now it is just hard coded
+	collection := client.Database("tenantdb").Collection("tenants")
```

Comment:
name of database and collection should either come from the configuration service or must be defined as constant on top of this file